test/test_torch.py

```diff
@@ -3361,7 +3361,7 @@ def test_conv_transposed_backward_agnostic_to_memory_format(self, device):
         input_.sum().backward()
 
     # TODO: this test should be in test_nn.py
-    @largeTensorTest('12GB')
+    @largeTensorTest('12GB', device='cuda')
```

Comment:
```
@onlyCUDA
@largeTensorTest('12GB')
```

if you want to only run the test on CUDA. I think passing `device='cuda'` to this decorator will just check if 12GB is available on the primary CUDA device, even when running on CPU. The check is implemented as:

```
_device = device if device is not None else self.get_primary_device()
            if not _has_sufficient_memory(_device, size_bytes):
```